### ISSUE TITLE
Why do you use the test data set to adjust the hyperparameters during the training phase instead of the validation data set?

### ISSUE THREAD

**Issue by donglongzi** [chenhaoxing/M2L#1]
There are two common problems as follows：
1. In the testing phase, you load the state of the model saved on the test set instead of the validation set. So is there a risk of overfitting the model on the test set?
2.  Why don’t you use the best state of the model on the validation set to evaluate the generalization performance of the model on the test set?
